Feature/user

alkafds opened the pull request.

alkafds committed:
added features to usercontroller

alkafds committed:
added delete function

alkafds committed:
added delete function

alkafds committed:
modified update

rza-git commented on routes/auth.js:
update pake put

rza-git commented on controllers/userController.js:
```diff
@@ -44,34 +45,100 @@ class UserController {
-      res.status(200).json(users)
-    } catch(err) {
-      next(err)
+      const userId = req.loggedInUser;
```
typo!!. Harusnya loggedUser

rza-git commented on routes/auth.js:
```diff
@@ -4,5 +4,9 @@ const UserController = require("../controllers/userController");
 
 router.post("/register", UserController.register);
 router.post("/login", UserController.login);
+router.post("/update", UserController.update);
```
kalo update pake put

rza-git replied:
```suggestion
router.put("/:id", UserController.update);
```

rza-git commented on controllers/userController.js:
```diff
@@ -44,34 +45,100 @@ class UserController {
       if (!comparePassword) {
         throw { name: "InvalidCredential" };
       }
-      const accessToken = signToken({
```
ini kenapa accessToken diubah lagi???

rza-git commented on controllers/userController.js:
```diff
@@ -44,34 +45,100 @@ class UserController {
+  }
+  static async getByID(req, res, next) {
+    try {
+      const userId = req.loggedInUser;
```
typo!!!, loggedUser

rza-git commented on controllers/userController.js:
```diff
@@ -44,34 +45,100 @@ class UserController {
-// AUTHENTICATED ROUTES
+  static async update(req, res, next) {
+    try {
+      const userId = req.loggedInUser;
```
typo

rza-git commented on controllers/userController.js:
```diff
@@ -26,7 +27,7 @@ class UserController {
       });
       res.status(201).json(user);
-      // console.log(user)
+      console.log(user)
```
console log apus aja

rza-git commented on controllers/userController.js:
```diff
@@ -68,12 +74,70 @@ class UserController {
-      const users = await User.findAll()
+      const { page = 1, limit = 10 } = req.query;
+      const offset = (page - 1) * limit;
+      const users = await User.findAndCountAll({
```
const {count, data} = await User.findAndCountAll({blabla})

rza-git commented on controllers/userController.js:
```diff
@@ -68,12 +74,70 @@ class UserController {
+
+  static async update(req, res, next) {
+    try {
+      const userId = req.params.id;
```
dapetin user id dari req.loggedUser

rza-git commented on routes/auth.js:
```diff
@@ -4,5 +4,9 @@ const UserController = require("../controllers/userController");
 router.post("/login", UserController.login);
+router.post("/update", UserController.update);
+router.get("/:id", UserController.getByID);
+router.post("/delete", UserController.delete);
```
```suggestion
router.delete("/:id", UserController.delete);
```

rza-git commented on controllers/userController.js:
```diff
@@ -44,34 +45,100 @@ class UserController {
-        role: user.role,
-      });
-      
+      let accessToken;
```
accessToken harus direvert kayak semula. Karena fitur login sudah selesai

alkafds committed:
Add files via upload

alkafds committed:
fixed features

- reverted register/login
- fixed typos
- fixed data type for role in user models
- fixed update, delete and getbyID features

alkafds committed:
Add files via upload

alkafds committed:
Added .env

alkafds committed:
Merge branch 'development' into feature/user
